### PR TITLE
Download unsubscribe request reports

### PR DESCRIPTION
**Pull request by BlessedDev**
This PR enables downloading an `unsubscribe-request-report` as described in [this](https://trello.com/c/BeZGTW0E/53-let-users-download-email-unsubscribe-requests) card.
The PR for the API endpoints can be found [here](https://github.com/alphagov/notifications-api/pull/4152).

### CODE REVIEW

File: app/main/views/unsubscribe_requests.py [hunk 1]
```diff
@@ -41,3 +43,59 @@ def unsubscribe_request_report(service_id, batch_id=None):
...
+            for row in report["unsubscribe_requests"]:
+                data.append([row[key] for key in column_names.keys()])
+            return (
+                Spreadsheet.from_rows(data).as_csv_data,
+                200,
+                {
+                    "Content-Type": "text/csv; charset=utf-8",
+                    "Content-Disposition": f'attachment; filename="Email unsubscribe requests '
+                    f"{format_date_normal(report['earliest_timestamp'])} "
+                    f"to {format_date_normal(report['latest_timestamp'])}.csv",
```
Comment (quis) [inline app/main/views/unsubscribe_requests.py:74]: Filenames should have numeric dates so they are listed in order once downloaded
```suggestion
                    f"{format_date_numeric(report['earliest_timestamp'])} "
                    f"to {format_date_numeric(report['latest_timestamp'])}.csv",
```

File: app/main/views/unsubscribe_requests.py [hunk 2]
```diff
@@ -41,3 +43,59 @@ def unsubscribe_request_report(service_id, batch_id=None):
...
+def download_unsubscribe_request_report(service_id, batch_id=None):
+    if not batch_id:
+        return redirect(url_for("main.create_unsubscribe_request_report", service_id=service_id))
+    else:
+        try:
+            report = service_api_client.get_unsubscribe_request_report(service_id, batch_id)
+            column_names = {
+                "email_address": "Email address",
+                "template_name": "Template name",
+                "original_file_name": "Contact list file name",
```
Comment (quis) [inline app/main/views/unsubscribe_requests.py:60]:
```suggestion
                "original_file_name": "Uploaded spreadsheet file name",
```

File: app/main/views/unsubscribe_requests.py [hunk 3]
```diff
@@ -41,3 +43,59 @@ def unsubscribe_request_report(service_id, batch_id=None):
...
+        created_report_data = service_api_client.create_unsubscribe_request_report(service_id, unbatched_report_data)
+        return redirect(
+            url_for(
+                "main.download_unsubscribe_request_report",
+                service_id=service_id,
+                batch_id=created_report_data["report_id"],
+            )
+        )
+    except HTTPError as e:
+        raise e
```
Comment (quis) [inline app/main/views/unsubscribe_requests.py:101]: Same point about `try/except` here

**Review by quis — CHANGES_REQUESTED** [review]

File: app/main/views/unsubscribe_requests.py [hunk 4]
```diff
@@ -41,3 +43,59 @@ def unsubscribe_request_report(service_id, batch_id=None):
...
+                200,
+                {
+                    "Content-Type": "text/csv; charset=utf-8",
+                    "Content-Disposition": f'attachment; filename="Email unsubscribe requests '
+                    f"{format_date_normal(report['earliest_timestamp'])} "
+                    f"to {format_date_normal(report['latest_timestamp'])}.csv",
+                },
+            )
+        except HTTPError as e:
+            raise e
```
[the remark this answers is not in the document]
Comment (BlessedDev) [reply app/main/views/unsubscribe_requests.py:78]: I had intended to capture specific `http 400` errors raised by the API  but changed my mind and forgot to adjust the code.

**Review by quis — APPROVED** [review]
Good to go once the API side is merged

**Pull request merged**